pkg/kn/commands/service/human_readable_flags.go

```diff
@@ -15,26 +15,40 @@
 package service
 
 import (
+	"sort"
+
 	"github.com/knative/client/pkg/kn/commands"
 	hprinters "github.com/knative/client/pkg/printers"
 	servingv1alpha1 "github.com/knative/serving/pkg/apis/serving/v1alpha1"
 	metav1beta1 "k8s.io/apimachinery/pkg/apis/meta/v1beta1"
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
+var baseKServiceColumnDefinitions = []metav1beta1.TableColumnDefinition{
+	{Name: "Name", Type: "string", Description: "Name of the Knative service."},
+	{Name: "Url", Type: "string", Description: "URL of the Knative service."},
+	//{Name: "LastCreatedRevision", Type: "string", Description: "Name of last revision created."},
+	//{Name: "LastReadyRevision", Type: "string", Description: "Name of last ready revision."},
+	{Name: "Generation", Type: "integer", Description: "Sequence number of 'Generation' of the service that was last processed by the controller."},
+	{Name: "Age", Type: "string", Description: "Age of the service."},
+	{Name: "Conditions", Type: "string", Description: "Conditions describing statuses of service components."},
+	{Name: "Ready", Type: "string", Description: "Ready condition status of the service."},
+	{Name: "Reason", Type: "string", Description: "Reason for non-ready condition of the service."},
```

Comment:
Take a look at priority field as done in kubectl, we could do something like
priority 0 = namespace
priority 1 = default
priority 2 = reserved for ` -o wide` option

if all-namespace flag is given, print priority 0,1
if all-namespace flag is not give, print priority 1

WDYT ?

Reply:
SGTM. I updated to use priority field for switching output of column definitions 👍